cmd/tikv-ctl/src/main.rs

```diff
@@ -776,9 +797,186 @@ fn compact_whole_cluster(
+                            region_ids.is_empty() || region_ids.contains(region_id)
+                        })
+                        .for_each(|(key_range, region_id)| {
+                            // Get key range from region.
+                            let key_range = build_key_range(&key_range.0, &key_range.1, false);
+
+                            let addr = addr.clone();
+                            let cfg_inner = cfg.clone();
+                            let mgr = Arc::clone(&mgr);
+                            let f = async move {
+                                let debug_executor =
+                                    new_debug_executor(&cfg_inner, None, Some(&addr), mgr);
+                                debug_executor.flashback_to_version(
+                                    version,
+                                    region_id,
+                                    key_range,
+                                    start_ts.into_inner(),
+                                    0,
+                                )
+                            };
+                            futures.push(f);
+                        })
+                });
+
+            // Wait for finishing prepare flashback.
+            match tokio::time::timeout(
+                Duration::from_secs(FLASHBACK_TIMEOUT),
+                try_join_all(futures),
+            )
+            .await
+            {
+                Ok(res) => {
+                    if let Err(key_range) = res {
+                        // Retry specific key range to prepare flashback.
+                        let retry_stores_region = load_leaders_to_each_store(
+                            &pd_client,
+                            key_range.get_start_key().to_vec(),
+                            key_range.get_end_key().to_vec(),
+                        );
+                        // Need to update `stores_leader` to replace stale key range.
+                        for (store_id, leaders) in retry_stores_region {
+                            let regions = stores_leader
+                                .entry(store_id)
+                                .or_insert_with(HashMap::default);
+                            regions.extend(leaders);
+                        }
+                        continue;
+                    }
+                    break;
+                }
+                Err(e) => {
+                    println!("prepare flashback timeout. err: {:?}", e);
+                    return;
+                }
+            }
+        }
+
+        // Start flashback for all regions.
+        let commit_ts = pd_client.get_tso().await.unwrap();
+        loop {
+            let mut futures = Vec::default();
+            stores_leader
+                .clone()
+                .into_iter()
+                .for_each(|(store_id, leaders)| {
+                    let addr = pd_client.get_store(store_id).unwrap().address;
+                    leaders
```

Comment:
can we know the rough process?

Reply:
Can check in metrics after https://github.com/tikv/tikv/pull/14774 merged